setupext.py

```diff
@@ -531,6 +540,38 @@ def do_custom_build(self):
         """
         pass
 
+    def install_help_msg(self):
+        """
+        The help message to show if the package is not installed. The help
+        message shown depends on whether some class variables are present.
+        """
+        def _try_managers(*managers):
+            for manager in managers:
+                pkg_name = self.pkg_names.get(manager, None)
+                if pkg_name:
+                    try:
+                        _ = check_output(["which", manager],
+                                         stderr=subprocess.STDOUT)
+                        return ('Try installing {0} with `{1} install {2}`'
+                                .format(self.name, manager, pkg_name))
+                    except subprocess.CalledProcessError:
+                        pass
+
+        message = None
+        if sys.platform == "win32":
+            url = self.pkg_names.get("windows_url", None)
+            if url:
+                message = ('Please check {0} for instructions to install {1}'
+                           .format(url, self.name))
+        elif sys.platform == "darwin":
+            message = _try_managers("brew", "port")
+        elif sys.platform.startswith("linux"):
+            release = platform.linux_distribution()[0].lower()
+            if release in ('debian', 'ubuntu'):
```

Comment:
The supported distros in that function are:
```

_supported_dists = (
    'SuSE', 'debian', 'fedora', 'redhat', 'centos',
    'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
    'UnitedLinux', 'turbolinux')
```
If this is run on a non-supported distro, it returns empty strings `("", "", "")`.

There are also some issues with Ubuntu btw ... Some Ubuntu systems are detected as "debian" - http://bugs.python.org/issue9514